src/test_common.rs

```diff
@@ -86,9 +76,64 @@ impl Element for [u8; 16] {
     }
 }
 
-pub fn get_vec_tree_from_slice<U: Unsigned>(
+/// Implementation of SHA-256 Hasher
+///
+/// It is used for testing purposes
+#[derive(Copy, Clone)]
+pub struct Sha256Hasher {
+    engine: Sha256,
+}
+
+impl Sha256Hasher {
+    pub fn new() -> Sha256Hasher {
+        Sha256Hasher {
+            engine: Sha256::new(),
+        }
+    }
+}
+
+impl Debug for Sha256Hasher {
+    fn fmt(&self, _f: &mut Formatter<'_>) -> fmt::Result {
+        todo!()
+    }
+}
+
+impl Default for Sha256Hasher {
+    fn default() -> Self {
+        Sha256Hasher::new()
+    }
+}
+
+impl Hasher for Sha256Hasher {
+    // FIXME: contract is broken by design
+    fn finish(&self) -> u64 {
+        unimplemented!()
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        self.engine.input(bytes)
+    }
+}
+
+impl Algorithm<Item> for Sha256Hasher {
+    fn hash(&mut self) -> Item {
+        let mut result: Item = Item::default();
```

Comment:
FYI (no need to change). You can also use `let mut result = Item::default()`, which doesn't specify the type. Or `let mut result: Item = Default::default()`, which doesn't specify the type when calling `default()`.

Reply:
Right, thanks! I just prefer to explicitly state all types always